### PR TITLE
feat(server): query filter of core support

### PR DESCRIPTION
**Pull request by hexaforce**
# Overview

## What I've done
In the following Project retrieval queries, only items with the coreSupport flag set to true will be returned:

* GetProjects
* GetStarredProjects
* GetDeletedProjects

Currently, projects created in the visualizer are set with coreSupport as true.

## How I tested
[e2e test](https://github.com/reearth/reearth-visualizer/pull/1231/files#diff-f7d123401a1a82a94205ef95de13237e750f27d1cce05443924ce5e3fec0e541R14)

## Memo
https://www.notion.so/eukarya/BE-Projects-query-needs-to-return-Visualizer-projects-only-12f16e0fb16580529d87c5d2a8081c5e

## Summary by CodeRabbit

- **New Features**
	- Enhanced project retrieval methods to include core support criteria for filtering projects by workspace.
	- Updated test functionality to encompass both project creation and retrieval.

- **Bug Fixes**
	- Improved logic in project retrieval methods to ensure only relevant projects are returned based on core support and deletion status.

### CODE REVIEW

File: server/e2e/gql_project_test.go [hunk 1]
```diff
@@ -9,90 +9,65 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestCreateProject(t *testing.T) {
+// go test -v -run TestCreateAndGetProject ./e2e/...
```
Comment (pyshx) [inline server/e2e/gql_project_test.go:12]: remove this

File: server/e2e/gql_project_test.go [hunk 2]
```diff
@@ -9,90 +9,65 @@ import (
...
-		true, baseSeeder)
+	}, true, baseSeeder)
+
+	testData(e)
+	// test1-1 => coreSupport:default deleted:false
+	// test1-2 => coreSupport:default deleted:true
+	// test2-1 => coreSupport:true    deleted:false
+	// test2-2 => coreSupport:true    deleted:true
+	// test3-1 => coreSupport:false   deleted:false
+	// test3-2 => coreSupport:false   deleted:true
```
Comment (pyshx) [inline server/e2e/gql_project_test.go:28]: Remove this

File: server/e2e/gql_project_test.go [hunk 3]
```diff
@@ -9,90 +9,65 @@ import (
...
-		Value("createProject").Object().
-		Value("project").Object().
-		// ValueEqual("id", pId.String()).
-		ValueEqual("name", "test").
-		ValueEqual("description", "abc").
-		ValueEqual("imageUrl", "").
-		ValueEqual("coreSupport", true)
 }
 
+// go test -v -run TestSortByName ./e2e/...
```
Comment (pyshx) [inline server/e2e/gql_project_test.go:62]: Not needed

**Review by pyshx — APPROVED** [review]

File: server/e2e/gql_project_test.go (hunk 2 above)
Comment (hexaforce) [reply server/e2e/gql_project_test.go:28]: yes

File: server/e2e/gql_project_test.go (hunk 3 above)
Comment (hexaforce) [reply server/e2e/gql_project_test.go:62]: yes

File: server/e2e/gql_project_test.go (hunk 1 above)
Comment (hexaforce) [reply server/e2e/gql_project_test.go]: yes

File: server/e2e/gql_project_test.go (hunk 2 above)
Comment (hexaforce) [reply server/e2e/gql_project_test.go]: yes

**Pull request merged**